pandas/tests/io/json/test_normalize.py

```diff
@@ -129,6 +129,12 @@ def test_value_array_record_prefix(self):
         expected = DataFrame([[1], [2]], columns=['Prefix.0'])
         tm.assert_frame_equal(result, expected)
 
+    def test_list_record_path(self):
+        # GH 21605
+        result = json_normalize({'A': {'B': [1, 2]}}, ['A', 'B'])
```

Comment:
I think this bug is different from #21608 .
`json_normalize` raises an error even if the array elements are objects. 
```
>>> json_normalize({'A': {'B': [{'X': 1}, {'X': 2}]}}, ['A', 'B'])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python36\lib\site-packages\pandas\io\json\normalize.py", line 257, in json_normalize
    _recursive_extract(data, record_path, {}, level=0)
  File "C:\Python36\lib\site-packages\pandas\io\json\normalize.py", line 230, in _recursive_extract
    seen_meta, level=level + 1)
  File "C:\Python36\lib\site-packages\pandas\io\json\normalize.py", line 233, in _recursive_extract
    recs = _pull_field(obj, path[0])
  File "C:\Python36\lib\site-packages\pandas\io\json\normalize.py", line 180, in _pull_field
    result = result[spec]
TypeError: string indices must be integers
```

Reply:
Understood the error in question here is different but my point is that by having the `record_path` point to a JSON array instead of numbers instead of objects you are implicitly introducing a dependency on the other change that adds no value to this test case. If you point the `record_path` to a JSON object which is already supported this can move independent of the other change.

Reply:
I added a minimal test case to trigger the bug. The same bug triggers if the array elements are objects. Sure, I can change the test case to the one in my previous comment.